snapcraft/_baseplugin.py

```diff
@@ -85,10 +85,11 @@ def __init__(self, name, options, project=None):
         self.project = project
         self.options = options
 
+        part_dir = name.replace('/', '\N{BIG SOLIDUS}')
```

Comment:
anyway, if you decide to keep the character that looks like slash but it's not, a comment is required here explaining why.